logical_switch.go

```diff
@@ -271,3 +271,57 @@ func (odbi *ovnDBImp) lslblistImp(lswitch string) ([]*LoadBalancer, error) {
 	}
 	return listLB, nil
 }
+
+func (odbi *ovnDBImp) lsExtIdsAddImp(lswitch string, external_ids map[string]string) (*OvnCommand, error) {
+	var operations []libovsdb.Operation
+	row := make(OVNRow)
+	row["name"] = lswitch
+	lsuuid := odbi.getRowUUID(tableLogicalSwitch, row)
+	if len(lsuuid) == 0 {
+		return nil, ErrorNotFound
+	}
+	if len(external_ids) == 0 {
+		return nil, nil
+	}
+	mutateSet, err := libovsdb.NewOvsMap(external_ids)
+	if err != nil {
+		return nil, err
+	}
+	mutation := libovsdb.NewMutation("external_ids", opInsert, mutateSet)
+	condition := libovsdb.NewCondition("name", "==", lswitch)
+	mutateOp := libovsdb.Operation{
+		Op:        opMutate,
+		Table:     tableLogicalSwitch,
+		Mutations: []interface{}{mutation},
+		Where:     []interface{}{condition},
+	}
+	operations = append(operations, mutateOp)
+	return &OvnCommand{operations, odbi, make([][]map[string]interface{}, len(operations))}, nil
+}
+
+func (odbi *ovnDBImp) lsExtIdsDelImp(lswitch string, external_ids map[string]string) (*OvnCommand, error) {
+	var operations []libovsdb.Operation
+	row := make(OVNRow)
+	row["name"] = lswitch
+	lsuuid := odbi.getRowUUID(tableLogicalSwitch, row)
+	if len(lsuuid) == 0 {
+		return nil, ErrorNotFound
+	}
+	if len(external_ids) == 0 {
```

Comment:
@hzhou8  is it correct?

Reply:
i think that adding additional method is overhead. We need to have method that receives struct of kind logical_switch or another, and getRowUUID based on this struct converted to map(already have such method). So we can fill any of fields and get back rows that matched.

Reply:
@vtolstov I agree in general we should make the interfaces more generic. We will see how your POC works. Before that is ready, we may still need to add methods for the functions we need. So I am ok with the approach of this patch for now.